sdk/core/Azure.Core/src/DynamicData/MutableJsonElement.cs

```diff
@@ -37,6 +37,9 @@ internal MutableJsonElement(MutableJsonDocument root, JsonElement element, strin
         /// </summary>
         public JsonValueKind? ValueKind
         {
+#if NET6_0_OR_GREATER
+            [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode("Calls JsonSerializer.SerializeToUtf8Bytes which requires unreferenced code.")]
```

Comment:
Note that for a `class` you can apply `RequiresUnreferencedCode` on the whole type. But since this is a `struct` that mechanism doesn't work.

cc @vitek-karas @MichalStrehovsky

Reply:
opened: https://github.com/dotnet/runtime/issues/90115